txn.go

```diff
@@ -167,41 +171,85 @@ func (txn *Txn) Insert(table string, obj interface{}) error {
 
 		// Handle the update by deleting from the index first
 		if update {
-			ok, val, err := indexSchema.Indexer.FromObject(existing)
+			switch indexer := indexSchema.Indexer.(type) {
+			case SingleIndexer:
+				ok, val, err := indexer.FromObject(existing)
```

Comment:
@armon updated in next commit. In my opinion the new version is more DRY but less clear, less "idiomatic Go".